### PR TITLE
Fixed XPATH for shoe size & shoe type, added shoe type & don’t quit parameters

### PR DESCRIPTION
**Pull request by alfredog1976**
Following @ ericruiz-developer login fixes, also fixed shoe size and type selection. Also added new parameters: shoe-type and dont-quit (if set, waits for user to hit Enter on command line, then driver.quit() called). Also, refreshes page if shoe size and type selection fails.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -68,12 +67,13 @@ def run(driver, username, password, url, shoe_size, login_time=None, release_tim
...
-
+            
             try:
-                select_shoe_size(driver=driver, shoe_size=shoe_size)
+                select_shoe_size(driver=driver, shoe_size=shoe_size, shoe_type=shoe_type)
             except Exception as e:
-                # If we fail to select shoe size, try to buy anyway
+                # Try refreshing page since you can't click Buy button without selecting size
                 LOGGER.exception("Failed to select shoe size: " + str(e))
+                continue
```
Comment (alexschimpf) [inline main.py:76]: I think my original intention was that if you've already set your shoe size preference in your profile (which is preferred), then this step may not even be required. But honestly, I'm not sure if that's even true.

File: main.py [hunk 2]
```diff
@@ -238,6 +237,8 @@ def wait_until_visible(driver, xpath=None, class_name=None, duration=10000, freq
     parser.add_argument("--select-payment", action="store_true")
     parser.add_argument("--purchase", action="store_true")
     parser.add_argument("--num-retries", type=int, default=1)
+    parser.add_argument("--dont-quit", action="store_true")
+    parser.add_argument("--shoe-type", default="M")
```
Comment (alexschimpf) [inline main.py:241]: I'd prolly just add `choices=("M", "F", "W", "Y")` here and be done with it. I wouldn't bother accepting all those variations you have below.

File: main.py [hunk 3]
```diff
@@ -147,25 +145,26 @@ def login(driver, username, password):
...
+def select_shoe_size(driver, shoe_size, shoe_type):
     LOGGER.info("Waiting for size dropdown to appear")
-    wait_until_visible(driver, class_name="expanded", duration=10)
+    wait_until_visible(driver, class_name="size-grid-button", duration=10)
 
     LOGGER.info("Selecting size from dropdown")
-    driver.find_element_by_class_name("expanded").find_element_by_xpath(
-        "//button[text()='{}']".format(shoe_size)).click()
+
+    if shoe_type == "Y" or shoe_type == "C":
```
Comment (alexschimpf) [inline main.py:160]: `shoe_type in ("Y", "C")` ... this is a pet peeve of mine 😄

File: main.py (hunk 1 above)
Comment (alfredog1976) [reply main.py:76]: I tested the other day with a release, I thought their ajax would update the DOM to display the "Buy $XXX" button (similar to what the SNKRs mobile app does) but that is not the case. Upper timeout could be up to 20 seconds before page refresh which could be critical for a hot new release. I can remove for now, if desired. Also thought about reducing the timeout for each to 5 seconds, I'll test that on next release. 

I'll test clicking the "Buy" button without shoe size selection, wasn't sure if that was a feature for the Nike site (works for SNKRs mobile app). Perhaps adding another parameter to skip shoe size/type selection if this is the case?

File: main.py (hunk 2 above)
Comment (alfredog1976) [reply main.py:241]: Sure no problem! I just figured from a user experience standpoint (user's who aren't technically inclined or forget what to input as a parameter) to just cover all the bases for them. I will remove. 

Also, just realized, some shoes only have the shoe size, so I'll have to somehow check before searching for the element and set the string accordingly. I'll work on it and include in a future update.

File: main.py (hunk 3 above)
Comment (alfredog1976) [reply main.py:160]: No problem, i'll update this. Great overall work by the way! :)

File: main.py (hunk 1 above)
Comment (ericruiz-developer) [reply main.py:76]: @alfredog1976 I've noticed the Nike is now just popping up a modal with an Error. Have you experienced this? I was curious if their bot protection is picking something up here.

Comment (alfredog1976) [reply main.py:76]: @ericruiz-developer I've noticed that too in my testing when I'm manually signing in. Refreshing the page 1 or more times seems to alleviate it. Haven't seen that error when running the bot though.

**Comment by lotannau4** [comment]
Unsubscribe

**Comment by ericruiz-developer** [comment]
Sorry to piggy back off your PR but I wanted to get eyes on this. Do you all have a discord? I'd be interested in pushing this thing forward and maybe even develop support for footsites. 

Either one of you interested in talking? @alfredog1976 @alexschimpf

File: main.py (hunk 1 above)
Comment (alfredog1976) [reply main.py:77]: This does happen with bot. One possible way to fix, create another loop for login logic and add a shorter timeout (10-20 seconds) at the myAccount:greeting checkpoint to force a refresh.

**Comment by alfredog1976** [comment]
> Sorry to piggy back off your PR but I wanted to get eyes on this. Do you all have a discord? I'd be interested in pushing this thing forward and maybe even develop support for footsites.
> 
> Either one of you interested in talking? @alfredog1976 @alexschimpf

my discord is the same: alfredog1976

File: main.py (hunk 1 above)
Comment (ericruiz-developer) [reply main.py:77]: Yeah, I actually haven't had the error when browsing manually. Only with the bot. I've been looking into it and it seems sites might be picking up the browser drivers. I was looking at altering the binary but if a refresh fixes it that would be waaaaay easier, I wonder if you'll be stuck refreshing every page though...

**Comment by ericruiz-developer** [comment]
> > Sorry to piggy back off your PR but I wanted to get eyes on this. Do you all have a discord? I'd be interested in pushing this thing forward and maybe even develop support for footsites.
> > Either one of you interested in talking? @alfredog1976 @alexschimpf
> 
> my discord is the same: alfredog1976

I need your 4 digits after your name too haha. Usually is like alfredog1976#1234 or something.

**Comment by sarthaktexas** [comment]
I’d like to help too! Kosmos#0688!
-- 
Thanks,
Sarthak Mohanty

Per Aspera Ad Astra

**Comment by alexschimpf** [comment]
Here ya go: alexschimpf#8874

File: main.py (hunk 2 above)
Comment (alexschimpf) [reply main.py:250]: If you add `choices=("M", "F", "W", "Y")` as a param for `add_argument`, the user will get an error if they use something outside of those accepted values.

File: main.py (hunk 1 above)
Comment (alexschimpf) [reply main.py:77]: I guess that is a fix for a separate PR then? We prolly need to discuss it a little more on Discord.

**Comment by alfredog1976** [comment]
> Sorry to piggy back off your PR but I wanted to get eyes on this. Do you all have a discord? I'd be interested in pushing this thing forward and maybe even develop support for footsites.
> 
> Either one of you interested in talking? @alfredog1976 @alexschimpf

oops sorry alfredog1976#0465

**Comment by alexschimpf** [comment]
@alfredog1976 Got some merge conflicts to resolve

**Review by alexschimpf — APPROVED** [review]

**Pull request merged**